Processing/Explore_New_Parks.Rmd

```diff
@@ -0,0 +1,303 @@
+    "#6eb2e4", 
+   "#459a74", 
+  "#eee461", 
+   "#3070ad", 
+  "#c66426", 
+   "#c07da5",
+  "#ECD8E4"
+)
+
+```
+
+# Input water supply table points
+
+```{r}
+
+table_path <- "data/Water_Supply_Systems/NPS_Water_Systems_Database.xlsx"
+
+
+
+# Supply table (Currently filtered to BRCA)
+supply_table <- read_excel(table_path, sheet = 2, na = "NA") %>%
+  janitor::clean_names() %>%
+  dplyr::filter(region %in% c("Pacific West Region", "Intermountain Region")) %>%
+  dplyr::filter(park_unit %in% parks)
+
+# Source table
+source_table <- read_excel(table_path, sheet = 3, na = "NA") %>%
+  janitor::clean_names() %>%
+  #dplyr::filter(state == "UT")
+  dplyr::filter(wsd_system_id %in% supply_table$wsd_system_id) %>%
+  dplyr::mutate(well_depth = as.numeric(well_depth))
+
+# Copy of source table as sf_object - some columns without location data are
+# dropped. Circle back to this later.
+source_table_locs <- source_table %>%
+  drop_na(c("source_longitude", "source_longitude")) %>%
+  st_as_sf(coords = (c("source_longitude","source_latitude")), 
+           crs = 4326,
+           remove = FALSE) 
+
+park_boundaries <- getParkSystem() %>%
+  dplyr::filter(UNIT_CODE %in% parks)
+```
+
+
+Plot all park boundaries and water supply points
+```{r}
+
+a <- mapview(park_boundaries,
+          col.regions = "seagreen",
+          alpha.regions = 0.3,
+          layer.name = "Park Boundaries") +
+  mapview(source_table_locs,
+        zcol = "source_type",
+        cex = 4,
+        layer.name = "Source Type") 
+
+  leafem::addStaticLabels(a,
+                        label = park_boundaries$UNIT_CODE,
+                  noHide = TRUE,
+                  direction = 'right',
+                  textOnly = TRUE,
+                  textsize = "15px",
+                  style = list("color" = "black", "font-weight" = "bold"))
+  
+
+```
+
+Get NWIS stream gauge and well data that falls within a 0.1 degree buffer of 
+parks. We specifically want to make sure data sites have some sort of time
+series that occur after 1980, so we'll filter to only include sites with > 10
+values and for which the end date is after 1980.
+
+```{r, eval = TRUE, message = FALSE, warning = FALSE}
+
+aoi <- park_boundaries %>% st_buffer(.1)
+
+mapview(aoi)
+
+unioned_aoi <- aoi #%>%
+  #st_union() %>%
+  #st_cast(., "POLYGON") %>%
+  #st_as_sf()
+
+mapview(unioned_aoi)
+ 
+gage_sites <- vector("list", length = nrow(unioned_aoi))
+  
+  for (i in 1:nrow(unioned_aoi)){
+    
+    bbox <- sf::st_bbox(unioned_aoi[i,]) %>%
+      as.vector() %>%
+      round(., digits=7) %>% 
+      paste(collapse = ",")
+    
+    gage_sites[[i]] <- dataRetrieval::whatNWISdata(bBox = bbox)
+    
+  }
+  
+  gage_sites <- dplyr::bind_rows(gage_sites) %>%
+    dplyr::filter(year(end_date) > 1980,
+                  count_nu > 10,
+                  data_type_cd %in% c("dv", "gw"),
+                  site_tp_cd %in% c("ST","GW")) %>%
+    sf::st_as_sf(coords=c('dec_long_va','dec_lat_va'),crs=4269) %>%
```

Comment:
Double checking if 4269 is the native CRS of the NWIS data? Normally decimal degrees are 4326..if the goal is for the output to be 4269 that would be a second transformation step.

Reply:
@ccmothes NWIS uses NAD83 so it should be 4269.  I pulled/modified those lines from the listNWIS function, which wasn't working for me at the time.  @kathryn-willi has since updated that function, but perhaps can confirm that I'm correct in 4269.  This RMD no longer contains those lines, but it does not utilize listNWIS which maintains 4269.

Reply:
Correct it's NAD83!